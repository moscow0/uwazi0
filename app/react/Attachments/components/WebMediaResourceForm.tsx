import { Field, LocalForm } from 'react-redux-form';
import { Translate } from 'app/I18N';
import { Icon } from 'UI';
import React from 'react';
import { FormGroup } from 'app/Forms';

interface WebMediaResourceFormProps {
  handleSubmit: (args: any) => void;
  dispatch?: (dispatch: Function) => void;
  url?: string | null;
  hasName?: boolean;
}

const WebMediaResourceForm = ({
  handleSubmit,
  url,
  dispatch,
  hasName = false,
}: WebMediaResourceFormProps) => {
  const validators = {
    ...(hasName && { name: { required: (val: any) => !!val && val.trim() !== '' } }),
    url: { required: (val: any) => !!val && val.trim() !== '' },
  };

  return (
    <LocalForm
      onSubmit={handleSubmit}
      getDispatch={dispatch}
      model="urlForm"
      validators={validators}
      initialState={{ url }}
      className={!hasName ? 'select-from-link' : ''}
    >
      <FormGroup className="has-feedback" model=".url">
        <Field model=".url">
          <input
            type="text"
            className="form-control web-attachment-url"
            placeholder="Paste URL here"
          />
        </Field>
      </FormGroup>
      {hasName && (
        <FormGroup className="form-group" model=".name">
          <Field model=".name" className="field">
            <input type="text" className="form-control web-attachment-name" placeholder="Title" />
          </Field>
        </FormGroup>
      )}
      <button type="submit" className="btn">
        <Icon icon="link" />
        &nbsp; <Translate>Add from URL</Translate>
      </button>
    </LocalForm>
  );
};

export { WebMediaResourceForm };
