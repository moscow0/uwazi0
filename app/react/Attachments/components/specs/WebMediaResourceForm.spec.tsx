import React from 'react';
import { LocalForm } from 'react-redux-form';
import { shallow, ShallowWrapper } from 'enzyme';
import { WebMediaResourceForm } from '../WebMediaResourceForm';

let component: ShallowWrapper;

const submitSpy = jest.fn();

const render = (name: boolean) => {
  component = shallow(<WebMediaResourceForm handleSubmit={submitSpy} hasName={name} />);
};

it('Should match render of wem media form', () => {
  render(false);
  expect(component).toMatchSnapshot();
});

it('should also display a name field when hasName is true', () => {
  render(true);
  expect(component).toMatchSnapshot();
});

it('should call the handleSubmit function on submit', () => {
  render(true);
  component.find(LocalForm).simulate('submit');
  expect(submitSpy).toHaveBeenCalled();
});
