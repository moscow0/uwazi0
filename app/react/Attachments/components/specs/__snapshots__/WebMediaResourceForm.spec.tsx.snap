// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match render of wem media form 1`] = `
<LocalForm
  className="select-from-link"
  initialState={
    Object {
      "url": undefined,
    }
  }
  model="urlForm"
  onSubmit={[MockFunction]}
  validators={
    Object {
      "url": Object {
        "required": [Function],
      },
    }
  }
>
  <Connect(FormGroup)
    className="has-feedback"
    model=".url"
  >
    <Modeled(Connect(Field))
      model=".url"
    >
      <input
        className="form-control web-attachment-url"
        placeholder="Paste URL here"
        type="text"
      />
    </Modeled(Connect(Field))>
  </Connect(FormGroup)>
  <button
    className="btn"
    type="submit"
  >
    <Connect(Icon)
      icon="link"
    />
      
    <Connect(Translate)>
      Add from URL
    </Connect(Translate)>
  </button>
</LocalForm>
`;

exports[`should also display a name field when hasName is true 1`] = `
<LocalForm
  className=""
  initialState={
    Object {
      "url": undefined,
    }
  }
  model="urlForm"
  onSubmit={[MockFunction]}
  validators={
    Object {
      "name": Object {
        "required": [Function],
      },
      "url": Object {
        "required": [Function],
      },
    }
  }
>
  <Connect(FormGroup)
    className="has-feedback"
    model=".url"
  >
    <Modeled(Connect(Field))
      model=".url"
    >
      <input
        className="form-control web-attachment-url"
        placeholder="Paste URL here"
        type="text"
      />
    </Modeled(Connect(Field))>
  </Connect(FormGroup)>
  <Connect(FormGroup)
    className="form-group"
    model=".name"
  >
    <Modeled(Connect(Field))
      className="field"
      model=".name"
    >
      <input
        className="form-control web-attachment-name"
        placeholder="Title"
        type="text"
      />
    </Modeled(Connect(Field))>
  </Connect(FormGroup)>
  <button
    className="btn"
    type="submit"
  >
    <Connect(Icon)
      icon="link"
    />
      
    <Connect(Translate)>
      Add from URL
    </Connect(Translate)>
  </button>
</LocalForm>
`;
